src/AggregateFunctions/AggregateFunctionGroupArraySorted.h

```diff
@@ -0,0 +1,390 @@
+#include <Functions/array/arraySort.h>
+
+#include <Common/Exception.h>
+#include <Common/ArenaAllocator.h>
+#include <Common/assert_cast.h>
+#include <Columns/ColumnConst.h>
+#include <DataTypes/IDataType.h>
+#include <base/sort.h>
+#include <Columns/IColumn.h>
+
+#include <AggregateFunctions/IAggregateFunction.h>
+
+#include <algorithm>
+#include <type_traits>
+#include <utility>
+
+#define AGGREGATE_FUNCTION_GROUP_ARRAY_MAX_ARRAY_SIZE 0xFFFFFF
+
+
+namespace DB
+{
+struct Settings;
+
+namespace ErrorCodes
+{
+    extern const int TOO_LARGE_ARRAY_SIZE;
+    extern const int INCORRECT_DATA;
+}
+
+template <typename T, typename Trait>
+class GroupArraySortedNumericImpl final
+    : public IAggregateFunctionDataHelper<GroupArrayNumericData<T, false>, GroupArraySortedNumericImpl<T, Trait>>
+{
+    using Data = GroupArrayNumericData<T, Trait::sampler != Sampler::NONE>;
+    static constexpr bool limit_num_elems = Trait::has_limit;
+    UInt64 max_elems;
+    UInt64 seed;
+
+public:
+    explicit GroupArraySortedNumericImpl(
+        const DataTypePtr & data_type_, const Array & parameters_, UInt64 max_elems_ = std::numeric_limits<UInt64>::max(), UInt64 seed_ = 123456)
+        : IAggregateFunctionDataHelper<GroupArrayNumericData<T, false>, GroupArraySortedNumericImpl<T, Trait>>(
+            {data_type_}, parameters_, std::make_shared<DataTypeArray>(data_type_))
+        , max_elems(max_elems_)
+        , seed(seed_)
+    {
+    }
+
+    String getName() const override { return "groupArraySorted"; }
+
+    void create(AggregateDataPtr __restrict place) const override /// NOLINT
+    {
+        [[maybe_unused]] auto a = new (place) Data;
+    }
+
+    void add(AggregateDataPtr __restrict place, const IColumn ** columns, size_t row_num, Arena * arena) const override
+    {
+        const auto & row_values = assert_cast<const ColumnVector<T> &>(*columns[0]).getData();
+
+        if (limit_num_elems && row_values.size() < max_elems)
+            throw Exception(ErrorCodes::INCORRECT_DATA, "The max size of result array is bigger than the actual array size");
+
+        const auto & row_value = row_values[row_num];
+        auto & cur_elems = this->data(place);
+
+        ++cur_elems.total_values;
+        cur_elems.value.push_back(row_value, arena);
+
+        if (!limit_num_elems && columns[0]->size() == cur_elems.value.size())
+            std::sort(cur_elems.value.begin(), cur_elems.value.end());
+        else if (columns[0]->size() == cur_elems.value.size())
+        {
+            /// To optimize, we sort (2 * max_size) elements of input array over and over again
+            /// and after each loop we delete the last half of sorted array
+            bool sorted = false, part_sorted = false;
+            while (!sorted)
+            {
+                if (cur_elems.value.size() >= max_elems * 2)
+                {
+                    std::sort(cur_elems.value.begin(), cur_elems.value.begin() + (max_elems * 2));
```

Comment:
Ok, but let's use `cur_elems.value.end()` instead of `max_elems * 2`. (cause `cur_elems.value.size()` can be > `max_elems * 2`)

Reply:
Also, as long as we know that values are numbers, let's use radix sort